savaslabs/savaslabs.github.io
# Issues #4676 and #4677: handle displaying and submitting comments

Fixes issues [#4676](https://pm.savaslabs.com/issues/4676) and [#4677](https://pm.savaslabs.com/issues/4677)

## Summary of changes

- Displays comments on a post
- Makes post request to comments server on form submission
- For now, prints the response from the server in the console

## Notes

I needed to lift the state up for:

- Making a new call to Squabble for comments when a new comment is submitted
- Hiding and showing the comment form (hide when comment is submitted, show initially if there are comments and anytime the "leave a comment" link is clicked)

Let me know if you have any suggestions for improving this code.

To do in a future PR:

- [ ] Move everything in the `app` directory of `_comments-app` to the root of `_comments-app`
- [ ] Success and error messages
- [ ] Comment count

## To test

- Start local comment server
- Pull this and run `npm install`
- `gulp clean && gulp serve`
- Try submitting a new comment. The comment should show and the form should disappear. I'll work on getting the comment to show in a more smooth manner when I do the issue for success/error messages.
- Test the "leave a comment" link. The one issue I haven't resolved is that after submitting a comment, if you click the link, it doesn't smooth-scroll you to the form like it should

### Pull request reviewer

As the reviewer, I have verified the following:
